dmscripts/send_stats_to_performance_platform.py

```diff
@@ -0,0 +1,163 @@
+    if res.status_code != 200:
+        logger.error(
+            u"Failed to send data: {code}: {cause}",
+            extra={'code': res.status_code, 'cause': res.json().get('message', res.text)}
+        )
+    return res.status_code
+
+
+def applications_by_stage(stats):
+    return _format_statistics(
+        stats,
+        'interested_suppliers',
+        {
+            'interested': {
+                'declaration_status': [None, 'started'],
+                'has_completed_services': False
+            },
+            'made-declaration': {
+                'declaration_status': 'complete',
+                'has_completed_services': False
+            },
+            'completed-services': {
+                'declaration_status': [None, 'started'],
+                'has_completed_services': True
+            },
+            'eligible': {
+                'declaration_status': 'complete',
+                'has_completed_services': True
+            }
+        }
+    )
+
+
+def services_by_lot(stats, framework):
+    return _format_statistics(
+        stats,
+        'services',
+        {
+            lot['slug']: {
+                'lot': lot['slug'],
+                'status': 'submitted',
+            } for lot in framework['lots']
+        }
+    )
+
+
+def send_by_stage_stats(stats, timestamp_string, period, pp_bearer):
+    data_type = "applications-by-stage"
+    processed_stats = applications_by_stage(stats)
+    data = [{
+        "_id": _generate_id(timestamp_string, period, data_type, stage),
+        "_timestamp": timestamp_string,
+        "service": "gcloud",
+        "stage": stage,
+        "count": processed_stats.get(stage, 0),
+        "dataType": data_type,
+        "period": period
+    } for stage in processed_stats]
+
+    return send_data(data, PERFORMANCE_PLATFORM_URLS[period]['stage'], pp_bearer)
+
+
+def send_by_lot_stats(stats, timestamp_string, period, framework, pp_bearer):
+    data_type = "applications-by-lot"
+    processed_stats = services_by_lot(stats, framework)
+    data = [{
+        "_id": _generate_id(timestamp_string, period, data_type, lot),
+        "_timestamp": timestamp_string,
+        "service": "gcloud",
+        "lot": lot,
+        "count": processed_stats.get(lot, 0),
+        "dataType": data_type,
+        "period": period
+    } for lot in processed_stats]
+
+    return send_data(data, PERFORMANCE_PLATFORM_URLS[period]['lot'], pp_bearer)
+
+
+@backoff.on_exception(backoff.expo, dmapiclient.HTTPError, max_tries=5)
+def send_framework_stats(data_api_client, framework_slug, period, pp_bearer):
+    stats = data_api_client.get_framework_stats(framework_slug)
+    framework = data_api_client.get_framework(framework_slug)['frameworks']
+    now = datetime.utcnow()
+    # _timestamp is the *start* of the period to which the data relates but the "reading" here is at the *end*
+    # of the period, so need to subtract the period from the current time
```

Comment:
are you sure this is the right approach to take? it seems that both approaches are as wrong as each other (it's not marking a time period at all, it's marking an infinitesimal point in time. I'm not sure doing this juggle is totally necessary and is possibly a little confusing/surprising...?